sdk/core/Azure.Core.TestFramework/src/ClientTestFixtureAttribute.cs

```diff
@@ -34,41 +54,93 @@ public IEnumerable<TestSuite> BuildFrom(ITypeInfo typeInfo, IPreFilter filter)
         {
             if (_serviceVersions.Any())
             {
-                foreach (object serviceVersion in _serviceVersions)
+                if (_additionalParameters?.Any() ?? false)
                 {
-                    var syncFixture = new TestFixtureAttribute(false, serviceVersion);
-                    var asyncFixture = new TestFixtureAttribute(true, serviceVersion);
-
-                    foreach (TestSuite testSuite in asyncFixture.BuildFrom(typeInfo, filter))
+                    foreach (var parameter in _additionalParameters)
```

Comment:
How does this work with recordings?  Just looking at the PR, I worry the per parameter recordings would stomp each other.

Reply:
Ugh, good one. We need to handle this in recording path creation.

Reply:
Good question - enabling this for existing recordings like I did in Tables, it happily read the existing recorded files, but I suppose it should have been looking for files named according to the testcase diff.

Reply:
We might want to use https://github.com/Azure/azure-sdk-for-net/pull/13430/files#diff-62cd6e88027f3a4e8fb8d64827568d8aR166 to set more metadata on the test context and then use it from RecordedTestBase. Something like "RecordingDirectory".